ce/api/percentileanomaly.py

```diff
@@ -292,64 +298,89 @@ def canonical_timestamp(timestamp, timescale, timeidx):
 
-        # calculate percentiles and anomalies
-        for p in projected_data:
-            # determine the baseline, if applicable
-            if calculate_anomaly:
-                baseline = None
-                date_format = '%Y-%m-%d %H:%M:%S'
-                projected_date = datetime.strptime(p["date"], date_format)
-
-                for b in baseline_data:
-                    baseline_date = datetime.strptime(b["date"], date_format)
-                    if p["timescale"] == b["timescale"] and baseline_date.month == projected_date.month:
-                        if not baseline:
-                            b["values"] =b["values"][0]
-                            baseline = float(b["values"])
-                        else:
-                            abort(500,
-                                    ("Multiple matching baseline datasets ",
-                                    "for {} {} {} {}").format(
-                                        baseline_model,
-                                        baseline_climatology,
-                                        p["timescale"], p["date"]))       
 
-                if(not baseline):
+        def determine_baseline(curr_idx, p_obj, baseline_data, baseline_model, baseline_climatology):
+            '''
+            The function determines the baseline value 
+            using the given arguments(p_obj and baseline_data).
+            It iterates through every b_obj in the baseline_data 
+            and checks if there is anything that corresponds to 
+            p_obj(a data object in projected_data). The 
+            corresponding b_obj's value is assigned to baseline.
+            If multiple or no matcing baseline values are found,
+            the function aborts the program.
+            '''
+            
+            baseline = None
+            b_obj = baseline_data[curr_idx]
+
+            if(b_obj is None):
                     abort(500, "Missing baseline data: {} {} {}".format(
-                                                baseline_model, baseline_climatology, p["timescale"]))    
+                            baseline_model, baseline_climatology, p_obj["timescale"]))    
 
             else:
-                baseline = 0.0
-            values = np.asarray([float(v) for v in p["values"]])
+                if not baseline and len(b_obj["values"]) == 1:
+                    b_obj["values"] = b_obj["values"][0]
+                    baseline = float(b_obj["values"])
+                else:
+                    abort(500,
+                            ("Multiple matching baseline datasets ",
+                            "for {} {} {} {}").format(
+                                baseline_model,
+                                baseline_climatology,
+                                p_obj["timescale"], p_obj["date"]))
+    
+            return baseline
+
+
+        # calculate percentiles and anomalies
+        curr_idx = 0
+        while curr_idx < len(projected_data):
```

Comment:
What we are doing here with `curr_idx` is a very common pattern, and Python provides a nicer tool for it than the `while` loop: We can (and should) use [`enumerate` ](https://docs.python.org/3.8/library/functions.html#enumerate) here. It's simpler and it's way harder to make a mistake with. Python is full of all kinds of goodies like this!